rmc/utils/data_loading.py

```diff
@@ -160,27 +167,51 @@ def create_transcript_ref(
     # 37 of which are on chrM
     ht = ht.filter(~ht.chrom.contains("M") & hl.is_defined(ht.preferred_transcript_id))
 
-    # Key by preferred transcript ID (MANE select or canonical)
-    # NOTE: All MANE select transcripts in VEP105/GENCODE39 are canonical
-    # (but not all canonical are MANE select)
-    ht = ht.key_by(transcript=ht.preferred_transcript_id)
-
-    # Filter transcript row annotation to canonical transcripts,
-    # rename symbol to hgnc_symbol, annotate with transcript version,
-    # and add 'chr' prefix to chrom if it doesn't exist
-    ht = ht.transmute(
-        transcripts=ht.transcripts.filter(lambda x: x.transcript_id == ht.transcript),
-        hgnc_symbol=ht.symbol,
-    )
-    transcript_len_check = ht.aggregate(hl.agg.count_where(hl.len(ht.transcripts) != 1))
-    if transcript_len_check > 0:
-        raise DataException(
-            "Transcript array length is not equal to 1 for"
-            f" {transcript_len_check} rows. Please double check!",
+    if filter_to_canonical:
+        # Key by preferred transcript ID (MANE select or canonical)
+        # NOTE: All MANE select transcripts in VEP105/GENCODE39 are canonical
+        # (but not all canonical are MANE select)
+        ht = ht.key_by(transcript=ht.preferred_transcript_id)
+
+        # Filter transcript row annotation to canonical transcripts,
+        # rename symbol to hgnc_symbol, annotate with transcript version,
+        # and add 'chr' prefix to chrom if it doesn't exist
+        ht = ht.transmute(
+            transcripts=ht.transcripts.filter(
+                lambda x: x.transcript_id == ht.transcript
+            ),
         )
+        transcript_len_check = ht.aggregate(
+            hl.agg.count_where(hl.len(ht.transcripts) != 1)
+        )
+        if transcript_len_check > 0:
+            raise DataException(
+                "Transcript array length is not equal to 1 for"
+                f" {transcript_len_check} rows. Please double check!",
+            )
+        ht = ht.annotate(
+            exons=ht.transcripts[0].exons,
+            transcript_version=ht.transcripts[0].transcript_version,
+            chrom=ht.transcripts[0].chrom,
+        )
+    else:
+        # Add additional boolean to track whether transcript ID is preferred ID
+        extra_annotation = "is_preferred_transcript"
```

Comment:
We don't actually use this in anything when `filter_to_canonical` is True — again not imperative, but this just won't be functional

Reply:
this table doesn't currently get referenced anywhere else `filter_to_canonical` gets set though

Reply:
I guess that's true. Honestly I don't want to sink much time into this functionality, can we just add TODOs or an issue wherever we see code is missing to execute this flag fully?

Reply:
added a note -- is there another spot you had in mind?